core/src/runtime/mod.rs

```diff
@@ -538,65 +549,78 @@ impl Runtime {
         match instruction.opcode {
             // Arithmetic instructions.
```

Comment:
We can increment a counter a single time with the opcode here but Im not sure how we would account for the syscalls